content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/check_service_retired.rb

```diff
@@ -1,37 +1,77 @@
-#
-# Description: This method checks to see that all of the service resources are retired before retiring the service.
-#
+module ManageIQ
+  module Automate
+    module Service
+      module Retirement
+        module StateMachines
+          module Methods
+            class CheckServiceRetired
+              def initialize(handle = $evm)
+                @handle = handle
+              end
 
-service = $evm.root['service']
-if service.nil?
-  $evm.log('error', "Service Object not found")
-  exit MIQ_ABORT
-end
+              def main
+                @handle.log('info', "Checking if all service resources have been retired.")
+                @result = 'ok'
+                check_all_service_resources
+              end
+
+              private
+
+              def service
+                @service ||= @handle.root['service'].tap do |obj|
+                  if obj.nil?
+                    @handle.log(:error, 'Service object not provided')
+                    raise 'Service object has not been provided'
+                  end
+                end
+              end
 
-$evm.log('info', "Checking if all service resources have been retired.")
+              def check_all_service_resources
+                service.service_resources.each do |sr|
+                  next if sr.resource.nil?
+                  next if sr.resource_type == "Service" && @handle.vmdb(:service).find(sr[:resource_id]).has_parent
+                  next unless sr.resource.respond_to?(:retired?)
+                  result = check_service_resource_retirement(sr)
+                  if result == 'retry'
+                    check_retired_state(result)
```

Comment:
@d-m-u Since we know that the result == retry, we can set the retry attributes and log message here. 
                 @handle.log('info', "Service: #{service.name} resource is not retired, setting retry.")
                 @handle.root['ae_result']         = 'retry'
                 @handle.root['ae_retry_interval'] = '1.minute'